cognite/neat/_client/_api/schema.py

```diff
@@ -48,3 +50,72 @@ def retrieve(
             views=views,
             containers=ContainerApplyDict(containers.as_write()),
         )
+
+    def retrieve_data_model_id(self, data_model_id: dm.DataModelIdentifier) -> DMSSchema:
+        data_models = self._client.data_modeling.data_models.retrieve(data_model_id, inline_views=True)
+        if len(data_models) == 0:
+            raise NeatValueError(f"Data model {data_model_id} not found")
+        data_model = data_models.latest_version()
+        return self.retrieve_data_model(data_model)
+
+    def retrieve_data_model(
+        self,
+        data_model: dm.DataModel[dm.View],
+    ) -> DMSSchema:
+        """Create a schema from a data model.
+
+        If a reference model is provided, the schema will include a reference schema. To determine which views,
+        and containers to put in the reference schema, the following rule is applied:
+
+            If a view or container space is different from the data model space,
+            it will be included in the reference schema.*
+
+        *One exception to this rule is if a view is directly referenced by the data model. In this case, the view will
+        be included in the data model schema, even if the space is different.
+
+        Args:
+            data_model: The data model to create the schema from.
+
+        Returns:
+            DMSSchema: The schema created from the data model.
+        """
+        views = dm.ViewList(data_model.views)
+        data_model_write = data_model.as_write()
+        data_model_write.views = list(views.as_ids())
+
+        container_ids = views.referenced_containers()
+        containers = self._client.loaders.containers.retrieve(list(container_ids), include_connected=True)
+
+        space_ids = [data_model.space]
+        space_read = self._client.loaders.spaces.retrieve(space_ids)
+        if len(space_read) != len(space_ids):
+            raise NeatValueError(f"Space(s) {space_read} not found")
+        space_write = space_read.as_write()
+
+        existing_view_ids = set(views.as_ids())
+        views = self._client.loaders.views.retrieve(
+            list(existing_view_ids), include_connected=True, include_ancestor=True
+        )
+
+        missing = existing_view_ids - set(views.as_ids())
```

Comment:
how can you have missing? what is the edge case? I would expect you will get same views back + any views that are implemented.